eng/common/scripts/modules/Package-Properties.psm1

```diff
@@ -0,0 +1,217 @@
+        {
+            return [PackageProps]::new($setupProps[0], $setupProps[1], $pkgPath)
+        }
+    }
+    return $null
+}
+
+function Extract-JavaPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "pom.xml"
+
+    if (Test-Path $projectPath)
+    {
+        $projectData = New-Object -TypeName XML
+        $projectData.load($projectPath)
+        $projectPkgName = $projectData.project.artifactId
+        $pkgVersion = $projectData.project.version
+
+        if ($projectPkgName.ToString() -eq $pkgName)
+        {
+            return [PackageProps]::new($pkgName, $pkgVersion.ToString(), $pkgPath)
+        }
+    }
+    return $null
+}
+
+# Takes package name and service Name
+# Returns important properties of the package as related to the language repo
+# Returns a PS Object with properties @ { pkgName, pkgVersion, pkgDirectoryPath, pkgReadMePath, pkgChangeLogPath }
+function Get-PkgProperties
+{
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [string]$PackageName,
+        [Parameter(Mandatory=$true)]
+        [string]$ServiceName,
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("net","java","js","python")]
+        [string]$Language,
+        [string]$RepoRoot="${PSScriptRoot}/../../../.."
+    )
+
+    $pkgDirectoryName = $null
+    $pkgDirectoryPath = $null
+    $serviceDirectoryPath = Join-Path $RepoRoot "sdk" $ServiceName
+    if (!(Test-Path $serviceDirectoryPath))
+    {
+        Write-Error "Service Directory $ServiceName does not exist"
+        exit 1
+    }
+
+    $directoriesPresent = Get-ChildItem $serviceDirectoryPath -Directory
+
+    foreach ($directory in $directoriesPresent)
+    {
+        $pkgDirectoryPath = Join-Path $serviceDirectoryPath $directory.Name
+        $pkgProps = Extract-PkgProps -pkgPath $pkgDirectoryPath -pkgName $PackageName
+        if ($pkgProps -ne $null)
+        {
+            return $pkgProps
+        }
+    }
+    Write-Error "Failed to retrive Properties for $PackageName"
+}
+
+# Takes ServiceName, Language, and Repo Root Directory
+# Returns important properties for each package in the specified service, or entire repo if the serviceName is not specified
+# Returns an array of PS Object with properties @ { pkgName, pkgVersion, pkgDirectoryPath, pkgReadMePath, pkgChangeLogPath }
+function Get-AllPkgProperties
+{
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("net","java","js","python")]
+        [string]$Language,
+        [string]$RepoRoot="${PSScriptRoot}/../../../..",
+        [string]$ServiceName=$null
+    )
+
+    if ([string]::IsNullOrEmpty($ServiceName)) {
+        $searchDir = Join-Path $RepoRoot "sdk"
+
+        foreach ($dir in (Get-ChildItem $searchDir -Directory))
+        {
+            $serviceDir = Join-Path $searchDir $dir.Name
+
+            if (Test-Path (Join-Path $serviceDir "ci.yml"))
+            {
+                $activePkgList = Get-PkgListFromYml -ciYmlPath (Join-Path $serviceDir "ci.yml")
```

Comment:
At some point we might also want to consider all child directories under the service folder as that might be different then what is in the ci.yml.  When we start having multiple ci.yml files for different purposes or we have cases were we are using these modules to actually validate or update the ci.yml files we might need to use the directories instead. For now lets stick with the ci.yml approach but this is something to keep in mind.